design/ux/Obscuro_Gateway.md

```diff
@@ -0,0 +1,264 @@
+    Alice -> MM: Automatically add "Obscuro" network with RPC\n"https://gateway.obscuro.network/v1?u=$UserId"
+end
+
+group Second click
+    Alice -> MM : Connect wallet and switch to Obscuro
+end
+
+group Third click
+    Alice -> MM: Automatically open MM with request to \nsign over "Register $UserId for $Acct"
+    note right
+        This text will be sent as is
+        accompanied by the signature and
+        will be parsed inside the enclave
+    end note
+    Alice -> MM : Confirm signature
+end
+
+Alice -> OG: All further Obscuro interactions will be to\nhttps://gateway.obscuro.network/v1?u=$UserId
+
+@enduml
+```
+
+The onboarding should be done in 3 clicks.
+1. The user goes to a website (like "obscuro.network"), where she clicks "Join Obscuro". This will add a network to their wallet.
+2. User connects the wallet to the page.
+3. In the wallet popup, the user has to sign over a message: "Register $UserId for $ACCT"
+
+##### Click 1
+1. Behind the scenes, a js functions calls "gateway.obscuro.network/v1/join" where it will generate a VK and send back the hash of the Public key. This is the "UserId" 
+2. After receiving the UserId, the js function will add a new network to the wallet.
+The RPC URL of the new Obscuro network will include the userid: "https://gateway.obscuro.network/v1?u=$UserId". 
+Notice that the UserId has to be included as a query parameter because it must be encrypted by https, as it is secret.
+
+##### Click 2
+After these actions are complete, the same page will now ask the user to connect the wallet and switch to Obscuro. 
+Automatically the page will open metamask and ask the user to sign over a text "Register $UserId for $ACCT", where ACCT is the current account selected in metamask.
+
+##### Click 3
+Once signed, this will be submitted in the background to: "https://gateway.obscuro.network/v1?u=$UserId&action=register"
+
+
+Note: Any further accounts will be registered similarly for the same UserId.
+
+Note: The user must guard the UserId. Anyone who can read it, will be able to read the data of this user.
+
+Todo - come up with a UX for revoking and generating a new HVK.
+
+The ultimate goal of this protocol is to submit the "Register $UserId for $ACCT" text to the gateway, which is required by an Obscuro node to authenticate viewing keys per address.
+
+Note: Alternative UXes that achieve the same goal are ok.
+
+
+## Multitenancy
+
+The curent WE is single-tenant. It assumes that all registered blockchain addresses belong to the same user.
+
+The OG will keep a many-to-one relationship between addresses and users. It will have multiple users, each with multiple addresses.
+
+Each request to the OG (except "/join") must have the "u" query parameter. 
+The first thing, the WE will lookup the userId and then operate in "Wallet Extension" mode, after loading all addresses.
+
+
+## HTTP Endpoints
+
+### Create User - GET "/join"
+
+- Generates a key-pair.
+- Hashes the public key - this is the UserId.
```

Comment:
Worth adding the PK of the VK. Maybe mentioning the VK-PK ? It got me confused, seemed like the UserId is the hash of the User's PK